sdk/storage/inc/common/memory_stream.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "http/stream.hpp"
+
+#include <vector>
+
+namespace Azure { namespace Storage {
+
+  class MemoryStream : public Azure::Core::Http::BodyStream {
+  public:
+    explicit MemoryStream(const char* data, std::size_t length) : m_data(data), m_length(length) {}
+
+    explicit MemoryStream(const std::vector<uint8_t>& data)
```

Comment:
The documentation on this should be extra careful since it only holds the reference of value in the customer's code logic, which might be destroyed after the memory stream initialization. It needs to point out that customers should manage the lifetime of this data vector until MemoryStream created with it is out of scope.